prereise/utility/translate_urban_rural_to_usa_tamu.py

```diff
@@ -0,0 +1,177 @@
+    sub2zone = sub2zone.reset_index(level="zone_id")
+
+    sub = grid.sub.join(sub2zone)
+    sub["zone"] = sub["zone_id"].replace(grid.id2zone)
+
+    sub = gpd.GeoDataFrame(sub, geometry=gpd.points_from_xy(sub.lon, sub.lat))
+    sub = sub[["zone", "geometry"]]
+
+    return sub
+
+
+def fix_lz_border(states, state, geometry):
+    """Takes a shape representing a loadzone and trims it to match state
+       borders. If there is only one loadzone in that state, we use the
+       original state border. If there are multiple loadzones in one state, we
+       trim the loadzone borders so they do not go outside the state border.
+
+    :param geopandas.geodataframe.GeoDataFrame states: GeoDataFrame of lower 48
+        US state borders with index = state, columns = ['geometry', 'state']
+    :param str state: Abbreviated name of the state.
+    :param TODO geometry: The shape of the loadzone to be trimmed
+    :return: (*TODO*) -- the modified loadzone shape
+    """
+
+    state_geom = states.loc[state, "geometry"]
+
+    if state in states_with_multiple_loadzones:
+        # Make sure loadzones don't go outside state borders
+        return geometry.intersection(state_geom)
+    else:
+        # Replace single loadzone states with state border shape
+        return state_geom
+
+
+def create_usa_tamu_convex_hull_shapefile(sub_gdf, states):
+    """Creates a Geodataframe that approximates the shapes of the USA tamu
+       loadzones. Loadzone shapes are made by taking the convex hull of all the
+       individual substations within that zone. A convex hull is the smallest
+       convex polygon containing all the points in each geometry.
+
+    :param geopandas.geodataframe.GeoDataFrame sub_gdf: GeoDataFrame of
+        substations with index = sub id, columns = ['geometry', 'zone']
+    :param geopandas.geodataframe.GeoDataFrame states: GeoDataFrame of lower 48
+        US state borders with index = state, columns = ['geometry', 'state']
+    :return: (*geopandas.geodataframe.GeoDataFrame*) -- gdf of USA tamu zones
+    """
+
+    tamu_loadzones = sub_gdf.dissolve("zone").convex_hull
+    tamu_loadzones = gpd.GeoDataFrame(tamu_loadzones)
+    tamu_loadzones = tamu_loadzones.rename(columns={0: "geometry"})
+
+    tamu_loadzones["state"] = tamu_loadzones.index
+    tamu_loadzones["state"] = tamu_loadzones["state"].replace(zone2state_abv)
+
+    # epsg:4269 matches urban_rural and is most commonly used by U.S. federal agencies.
+    # https://www.nceas.ucsb.edu/sites/default/files/2020-04/OverviewCoordinateReferenceSystems.pdf
+    tamu_loadzones = tamu_loadzones.set_crs("epsg:4269")
+
+    tamu_loadzones["geometry"] = tamu_loadzones.apply(
+        lambda x: fix_lz_border(states, x.state, x.geometry),
+        axis=1,
+    )
+
+    return tamu_loadzones
+
+
+def translate_urban_rural_to_usa_tamu():
+    """Downloads US state borders as well as census data for urban and rural
+       areas in the US. Creates a matrix that can be used to transform
```

Comment:
Was there supposed to be more after "Creates a matrix that can be used to transform"?